### PR TITLE
Add `cargo pgrx install --sudo`

### PR DESCRIPTION
**Pull request by eeeebbbbrrrr**
`cargo pgrx install --sudo` is a lightweight wrapper around `cargo pgrx package --release` that then uses `sudo cp $src $dest` to copy the extension file artifacts where `pg_config` thinks they should go on the system.

Typically, for package-manager installed Postgres installations, the extension target directories are owned by root and difficult for cargo-pgrx to deal with.  This gives the user a simple escape hatch for when they want to otherwise `cargo pgrx install`.

This idea came from @levkk over at PostgresML.

### CODE REVIEW

**Comment by eeeebbbbrrrr** [comment]
If we approve of this, I'll also cherry-pick it into the `v0.10-develop` branch and go ahead and publish it.

**Comment by eeeebbbbrrrr** [comment]
Screenshot 2023-10-05 at 10 06 16 AM

File: cargo-pgrx/src/command/sudo_install.rs
Comment (levkk) [inline cargo-pgrx/src/command/sudo_install.rs]: Can make it prettier, e.g.

```rust
println!("{} sudo cp {} {}", "       Running".bold().green(), src.display(), dest.display());
```

Comment (eeeebbbbrrrr) [reply cargo-pgrx/src/command/sudo_install.rs]: okay, will do.

File: cargo-pgrx/src/command/sudo_install.rs
Comment (thomcc) [inline cargo-pgrx/src/command/sudo_install.rs:59]: Wow does this ever look wrong (it's not wrong for this use case ofc, but still)

File: cargo-pgrx/src/command/sudo_install.rs
Comment (thomcc) [inline cargo-pgrx/src/command/sudo_install.rs:10]: I wonder if this would be better as `cargo pgrx install --sudo`. This is fine though.

I also slightly wonder if it's worth warning users who try to `sudo cargo pgrx install`, since it's both a footgun and a big security liability (there are definitely reasons to do it, so "warn" seems right rather than actually forbidding it). Neither here nor there though, just an idle thought.

File: cargo-pgrx/src/command/sudo_install.rs
Comment (thomcc) [inline cargo-pgrx/src/command/sudo_install.rs]: Using `output()` here will cause stderr/stdout to be piped. I'd expect that to prevent `sudo` from prompting the user for their password, or does that somehow work?

File: cargo-pgrx/src/command/sudo_install.rs
Comment (levkk) [reply cargo-pgrx/src/command/sudo_install.rs:10]: Most people won't even be able to because Rust is not installed as the root user. If you try, it will yell at you for...a while and ask you to set all kinds of environment variables before it will allow itself to be installed as root. But yes, I agree, a simple `if whoami == root then exit 1` in this case would not be a terrible idea.

Thinking out loud, `cargo pgrx install` would, by default never really work on any Linux system because for most people this would resolve to the installation in `/usr` . It works great on Mac because brew installs into `/opt/homebrew` where the user is allowed read/write access.

File: cargo-pgrx/src/command/sudo_install.rs
Comment (levkk) [reply cargo-pgrx/src/command/sudo_install.rs:59]: Yeah why don't you just use `std::fs::canonicalize` together with `shellexpand` as needed?

Comment (eeeebbbbrrrr) [reply cargo-pgrx/src/command/sudo_install.rs:59]:
```rust
let src = src.canonicalize()?;
let dest = make_absolute(src.strip_prefix(&outdir)?).canonicalize()?;
```

The `src.strip_prefix(&outdir)` call here includes the leading `/`.  So without `make_absolute()` the path ends up being relative to the current working directory.  Ask me how I know.

File: cargo-pgrx/src/command/sudo_install.rs
Comment (eeeebbbbrrrr) [reply cargo-pgrx/src/command/sudo_install.rs:10]: I'll rejigger this so that it is `cargo pgrx install --sudo`.  The spelling of this isn't a hill I'm willing to die on.

If we want a `if whoami == root` check, then we can do that as another PR.  That feels orthogonal to this.

File: cargo-pgrx/src/command/sudo_install.rs
Comment (eeeebbbbrrrr) [reply cargo-pgrx/src/command/sudo_install.rs]: It worked for me on Linux.  See the screenshot up above, but you see the sudo password prompt, can enter your password, and continue.

I wonder if sudo does something tricky if it detects stdout/err is piped?  I suppose I can spawn the process and just wait for it, tho.  I'll do that too.

**Comment by eeeebbbbrrrr** [comment]
Alrighty, I've cleaned things up in https://github.com/pgcentralfoundation/pgrx/pull/1322/commits/d9563c099be91f6c989d84a55de29a72cc998d39

~~CI is going to fail on pg16, but that's unrelated to this PR (and has been fixed/merged by #1326)~~

huh, it did not fail.  Even better!

**Comment by eeeebbbbrrrr** [comment]
@thomcc if I can get an approval on this today, that'd be great.  I want to put this behind us and get a new pgrx release out tomorrow.

**Review by thomcc — APPROVED** [review]
Looks good!

**Comment by eeeebbbbrrrr** [comment]
Thank you for updating the title/description, @workingjubilee.

**Pull request merged**
